cl/search/management/commands/cl_index_parent_and_child_docs.py

```diff
@@ -146,95 +330,228 @@ def handle(self, *args, **options):
                 q = queryset.iterator()
                 count = queryset.count()
-                self.process_queryset(
-                    q, count, SEARCH_TYPES.RECAP, pk_offset, document_type
-                )
+
             case SEARCH_TYPES.OPINION:
                 if document_type == "child":
                     queryset = (
                         Opinion.objects.filter(pk__gte=pk_offset)
                         .order_by("pk")
                         .values_list("pk", flat=True)
                     )
+                    task_to_use = "index_parent_or_child_docs"
                 else:
                     # Get Opinion Clusters objects by pk_offset.
                     queryset = (
                         OpinionCluster.objects.filter(pk__gte=pk_offset)
                         .order_by("pk")
                         .values_list("pk", flat=True)
                     )
+                    task_to_use = "index_parent_and_child_docs"
+                    if document_type == "parent":
+                        task_to_use = "index_parent_or_child_docs"
 
                 q = queryset.iterator()
                 count = queryset.count()
-                self.process_queryset(
-                    q, count, SEARCH_TYPES.OPINION, pk_offset, document_type
-                )
+
+            case _:
+                return
+
+        self.process_queryset(q, count, search_type, chunk_size, task_to_use)
 
     def process_queryset(
         self,
-        iterable: Iterable,
-        count: int,
-        search_type: str,
-        pk_offset: int,
-        document_type: str | None = None,
-    ) -> None:
+        iterable,
+        count,
+        search_type,
+        chunk_size,
+        task_to_use: str,
+    ):
+        event_doc_type = EventTable.get_member(
+            self.options.get("update_from_event_tables", None)
+        )
         queue = self.options["queue"]
-        chunk_size = self.options["chunk_size"]
         testing_mode = self.options.get("testing_mode", False)
+        pk_offset = self.options["pk_offset"]
+        document_type = self.options.get("document_type", None)
+
+        fields_map = {}
+        if event_doc_type == EventTable.DOCKET:
+            fields_map = recap_document_field_mapping["save"][Docket][
+                "docket_entry__docket"
+            ]
+            document_type = "parent"
+
+        if event_doc_type == EventTable.DOCKET_ENTRY:
+            fields_map = recap_document_field_mapping["save"][DocketEntry][
+                "docket_entry"
+            ]
+
+        if event_doc_type == EventTable.RECAP_DOC:
+            fields_map = recap_document_field_mapping["save"][RECAPDocument][
+                "self"
+            ]
+            document_type = "child"
 
         chunk = []
         processed_count = 0
         throttle = CeleryThrottle(queue_name=queue)
         # Indexing Parent and their child documents.
-        for item_id in iterable:
+        for item in iterable:
```

Comment:
Here, and I believe elsewhere these should be:

```suggestion
        for item in items:
```

Reply:
Sure, changed it.